config.ru

```diff
@@ -22,6 +22,7 @@ use Rack::Rewrite do
   rewrite    %r{(^.+)}, "$1.html", if: Proc.new {|env|
     env["REQUEST_PATH"].start_with?("/tagged")
   }
+  r302       %r{^\/(?!articles)(?!assets)(?!css)([^\/]+\/)*([^\/]+){1}[\/]?$}, '/articles/$2/'
```

Comment:
Use 301 code please. See https://moz.com/learn/seo/redirection for more explanations